truffle/src/main/java/org/truffleruby/core/module/ConstantLookupResult.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 1.0
+ * GNU General Public License version 2
+ * GNU Lesser General Public License version 2.1
+ */
+package org.truffleruby.core.module;
+
+import java.util.ArrayList;
+
+import org.truffleruby.RubyContext;
+import org.truffleruby.language.LexicalScope;
+import org.truffleruby.language.RubyConstant;
+
+import com.oracle.truffle.api.Assumption;
+import com.oracle.truffle.api.object.DynamicObject;
+
+public class ConstantLookupResult {
+
+    private final RubyConstant constant;
+    private final Assumption[] assumptions;
+
+    public ConstantLookupResult(RubyConstant constant, Assumption assumption) {
+        this.constant = constant;
+        this.assumptions = new Assumption[]{ assumption };
+    }
+
+    public ConstantLookupResult(RubyConstant constant, ArrayList<Assumption> assumptions) {
```

Comment:
Fixed in 797c8d3fc7798952e37de28ca8b0fbfcf99c4145 with a helper method to avoid too much noise.